.golangci.yml

```diff
@@ -124,3 +124,15 @@ issues:
     - text: 'SA1019: libvmi.WithRunning'
       linters:
         - staticcheck
+    # We should steer users towards client.VirtualMachine(vm.Namespace).start
+    - text: 'SA1019: tests.StartVirtualMachine'
+      linters:
+        - staticcheck
+    # We should steer users towards client.VirtualMachine(vm.Namespace).Stop
+    - text: 'SA1019: tests.StopVirtualMachine'
+      linters:
+        - staticcheck
+    # We should steer users towards client.VirtualMachine(vm.Namespace).Stop
+    - text: 'SA1019: tests.StopVirtualMachineWithTimeout'
+      linters:
+        - staticcheck
```

Comment:
We want a PRs to fail if they would use deprecated function. Let's not add them here if we do deprecate these functions.

Reply:
But otherwise we have the linter lane failing and i saw that this is [how we handle this usage](https://github.com/kubevirt/kubevirt/pull/12023/commits/752a550cb501b61fa832a03c7b454fc68c1c6f0e), is there another way?

Reply:
We should have override the context on that PR and then everything would work as expected. @EdDev we should revert the exclusion

Reply:
I do not understand what
> override the context on that PR and then everything would work as expected

means.

The deprecation comment may signal an author or reviewer to avoid using the function (some editors help with that). If the linter identifies that the function is in use in a covered codebase, it will fail.

While I prefer to “fix” the usage in one PR, I do not think it is correct to force it on the author. There are advantages of doing this in small steps. In fact, the whole point of marking it as deprecated hints that we do not solve everything now.

So, I am unsure what is your suggestion here. Is there another alternative?

Reply:
I did not check the meaning of configuration but what it sounds like is that in https://github.com/kubevirt/kubevirt/pull/12023/commits/752a550cb501b61fa832a03c7b454fc68c1c6f0e we silence these warnings

Reply:
It silents the linter so it will bot fail the check on the codebase that it covers.
I think that makes sense for the same reasoning mentioned in the prev comment.

Reply:
My bad @EdDev , I thought the check is smart and it would only detect new usage of the function. Now I see it just fails all the time, shame